### DISCUSSION TITLE
Energy consumption between last ON and OFF.

### DISCUSSION THREAD

**Discussion by UweClemens** [arendst/Tasmota#22133] [Ideas]
Hi, it would be a nice thing to have a sesnor for energy consumtion off the last ON/OFF sequence of a power switch like NOUS A1T like energy consumtion sensors already existing (today,yesterday).

**Comment by sfromis** [comment]
You already have the data, as the energy monitoring can produce current accumulated energy consumptio; subtract the value taken when/before powering on from the value when/after powering off. Could be done with rules, or backend tools.

**Comment by UweClemens** [comment]
That's right, but i think it should be no problem to add one sensor, because everthing is on board. So there would be no need for calculation and rules on backend tools.

**Reply by sfromis** [reply]
By "rules", I did not mean on your backend (that was just an additional option), but Tasmota rules.

**Reply by UweClemens** [reply]
Ah ok, didn't know about that.
Do you have experience using rules ?
It would be nice to see how I can make this power measurement with rules.
And how can I "programm" this rules on my device ?

**Reply by sfromis** [reply]
You have a learning curve then.... https://tasmota.github.io/docs/Rules/
That page includes "cookbook" examples of things done with rules.
